Mathlib/Data/Fin/Basic.lean

```diff
@@ -421,13 +421,13 @@ theorem rev_ne_iff {i j : Fin n} : rev i ≠ j ↔ i ≠ rev j := rev_eq_iff.not
 #align fin.last_val Fin.val_last
 #align fin.le_last Fin.le_last
 
-instance : BoundedOrder (Fin (n + 1)) where
-  top := last n
-  le_top := le_last
+instance [NeZero n] : BoundedOrder (Fin n) where
+  top := rev 0
```

Comment:
Remind me why we can't generalize `last` instead?

Reply:
It's harder because in Std. We would need to upstream `NeZero`. Certainly this information should be recorded somewhere in this file

Reply:
I had a go at generalising it and it turned out to be somewhat annoying to find precisely the right definition. We might also want to think about adjusting the definition of `Fin.ofNat'`, also. But essentially you are going to want to use either `Nat.pred` or `Fin.pred` in some way.

Reply:
I think there's also an argument that an updated `Fin.last` should not take an argument (it should just come from its type).